src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -265,6 +267,59 @@ public boolean canRecipeBeAdded(final IToken<?> ignored)
         return Optional.empty();
     }
 
+    /**
+     * Has a chance to reduce the resource requirements for the recipe in this building
+     * @param recipe the recipe we're possibly improving
+     * @param count the number of items (chances) 
+     * @param citizen The citizen, as the primary skill can improve the chances
+     */
+    public void improveRecipe(IRecipeStorage recipe, int count, ICitizenData citizen)
+    {
+        final double baseChance = 0.0625;
```

Comment:
could use some more constants for the numbers and strings used, instead of "minecolonies" use MOD_ID

Reply:
Where is MOD_ID defined, so I add the right Import?

Reply:
import static com.minecolonies.api.util.constant.Constants.MOD_ID;

Reply:
doesnt VS code find that? :D

Reply:
It didn't find it on it's own, since we already had a minecraft constants included. I found it by looking at other references.